Fixed length issue

BT-hstoeckle opened the pull request:
Fixed length issue in ESR fields, max 23 characters of name and address are take to not overlap with layout.

Commit:
Update payment_slip.py

BT-astauder commented:
👍 +1

open-net-sarl commented:
👍 +1

yvaucher commented on l10n_ch_payment_slip/models/payment_slip.py:
```diff
@@ -466,12 +466,12 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.setTextOrigin(x, y)
         text.setFont(font.name, font.size)
-        text.textOut(com_partner.name)
+        text.textOut(com_partner.name[:23])
```
I'm afraid in that case that we would loose valuable informations.
Let say I have a long street name, I could loose the street number.

Plus, with a different font size of 9 you can put 30 characters

Maybe we could make add an option to compute the font size depending on number of characters of the longest line of the address.

It would be:
longest_line <= 23 = size 11
longest_line <= 27 = size 10
longest_line <= 30 = size 9
longest_line <= 34 = size 8

according to my tests.

Commit:
Update payment_slip.py

Commit:
Update payment_slip.py

Font size will be adjusted based on longest line in com_partner name and address

Commit:
Update payment_slip.py

yvaucher commented:
Thanks for the improvement, could you also solve the conflict by doing a rebase ?

Commit:
Merge branch '9.0' into patch-1

yvaucher commented on l10n_ch_payment_slip/models/payment_slip.py:
```diff
@@ -481,16 +481,40 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
+        # we are moving in the original font size to new position
         text.moveCursor(0.0, font.size)
-        [text.textLine(l) for l in self._get_address_lines(com_partner) if l]
+        [text.textLine(l[:cutoff_length]) 
```
l10n_ch_payment_slip/models/payment_slip.py:515:42: W291 trailing whitespace

Commit:
Fixed W291 trailing whitespace in line 515

yvaucher commented:
Squashed commit and changed commit message at https://github.com/OCA/l10n-switzerland/commit/e0b57d53cce6c15f5403fa05b0d07888ee172760
